2012-01-17  Jakub Jelinek  <jakub@redhat.com>

	PR target/51876
	* config/arm/neon.md (*neon_vswp<mode>): Fix up operand
	numbers in the insn pattern.

--- gcc/config/arm/neon.md.jj	2012-01-13 21:47:35.000000000 +0100
+++ gcc/config/arm/neon.md	2012-01-17 12:46:14.419950640 +0100
@@ -2869,7 +2869,7 @@ (define_insn "*neon_vswp<mode>"
 	(match_operand:VDQX 1 "s_register_operand" "+w"))
    (set (match_dup 1) (match_dup 0))]
   "TARGET_NEON && reload_completed"
-  "vswp\t%<V_reg>1, %<V_reg>2"
+  "vswp\t%<V_reg>0, %<V_reg>1"
   [(set (attr "neon_type")
 	(if_then_else (match_test "<Is_d_reg>")
 		      (const_string "neon_bp_simple")
